cmd/soroban-rpc/internal/config/config_option.go

```diff
@@ -0,0 +1,78 @@
+package config
+
+import (
+	"fmt"
+	"go/types"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/stellar/go/support/errors"
+)
+
+// ConfigOptions is a group of ConfigOptions that can be for convenience
+// initialized and set at the same time.
+type ConfigOptions []*ConfigOption
+
+// Validate all the config options.
+func (options ConfigOptions) Validate() error {
+	for _, option := range options {
+		if option.Validate != nil {
+			err := option.Validate(option)
+			if err != nil {
+				return errors.Wrap(err, fmt.Sprintf("Invalid config value for %s", option.Name))
+			}
+		}
+	}
+	return nil
+}
+
+// ConfigOption is a complete description of the configuration of a command line option
+type ConfigOption struct {
+	Name           string                                 // e.g. "database-url"
+	EnvVar         string                                 // e.g. "DATABASE_URL". Defaults to uppercase/underscore representation of name
+	TomlKey        string                                 // e.g. "DATABASE_URL". Defaults to uppercase/underscore representation of name. - to omit from toml
+	Usage          string                                 // Help text
+	OptType        types.BasicKind                        // The type of this option, e.g. types.Bool
+	DefaultValue   interface{}                            // A default if no option is provided. Omit or set to `nil` if no default
+	ConfigKey      interface{}                            // Pointer to the final key in the linked Config struct
+	CustomSetValue func(*ConfigOption, interface{}) error // Optional function for custom validation/transformation
+	Validate       func(*ConfigOption) error              // Function called after loading all options, to validate the configuration
+	MarshalTOML    func(*ConfigOption) (interface{}, error)
+}
+
+// Returns false if this option is omitted in the toml
+func (o ConfigOption) getTomlKey() (string, bool) {
+	if o.TomlKey != "" {
+		return o.TomlKey, true
+	}
+	if o.EnvVar != "" && o.EnvVar != "-" {
+		return o.EnvVar, true
+	}
+	return strings.ToUpper(strings.ReplaceAll(o.Name, "-", "_")), true
```

Comment:
if the name isn't alphanumeric or contain "-"/"_", then the method should return false.
this case be done in a subsequent PR.